AndroidSDKCore/src/main/java/com/leanplum/internal/Util.java

```diff
@@ -660,7 +662,12 @@ public static void saveResponse(URLConnection op, OutputStream outputStream) thr
   private static String getResponse(HttpURLConnection op) throws IOException {
     InputStream inputStream;
     if (op.getResponseCode() < 400) {
-      inputStream = op.getInputStream();
+      String contentHeader = op.getHeaderField("content-encoding");
+      if (contentHeader != null && contentHeader.equalsIgnoreCase(Constants.LEANPLUM_SUPPORTED_ENCODING)) {
+        inputStream = new GZIPInputStream(op.getInputStream());
```

Comment:
What happens here if the server returns plain text but with the `content-encoding: gzip` response header?
What happens if the content is compressed, but the header is missing?

Reply:
1. What happens here if the server returns plain text but with the `content-encoding: gzip` response header?
A: Code will crash.As we discussed yesterday, the server has to respect the contract.

2. What happens if the content is compressed, but the header is missing?
A: Code will crash.As we discussed yesterday, the server has to respect the contract.